modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/tx/TransactionManager.java

```diff
@@ -67,7 +90,117 @@ public Transaction txStart(TransactionConcurrency concurrency, TransactionIsolat
      *
      * @return Transaction.
      */
-    public Transaction txStart() {
+    private Transaction txStart() {
         return txStart(PESSIMISTIC, REPEATABLE_READ);
     }
+
+    /**
+     * @param name Starter name.
+     * @param starter Caches starter.
+     */
+    public void registerStarter(String name, Runnable starter) {
+        cachesStarters.putIfAbsent(name, starter);
+
+        starter.run();
+    }
+
+    /**
+     * Recreate all caches by executing all registered starters.
+     */
+    public void recreateCaches() {
+        cachesStarters.forEach((name, starter) -> {
+            log.info("Creating caches for: {}", name);
+
+            starter.run();
+        });
+    }
+
+    /**
+     * @param action Action to execute.
+     * @return Action result.
+     */
+    private <T> T doInTransaction0(Supplier<T> action) {
+        try (Transaction tx = txStart()) {
+            T res = action.get();
+
+            tx.commit();
+
+            return res;
+        }
+    }
+
+    /**
+     * Execute action in transaction.
+     *
+     * @param name Action name.
+     * @param action Action to execute.
+     * @return Action result.
+     */
+    public <T> T doInTransaction(String name, Supplier<T> action) {
```

Comment:
Why you use Supplier when you code throw exception? Use Callable instead.

Reply:
"Callable.call() throws Exception" and that a pain to handle it.

Reply:
fixed